tests/functional/testSettings.js

```diff
@@ -886,4 +886,196 @@ describe('Settings', function() {
+        await vpn.waitForQuery(queries.screenSettings.STACKVIEW.ready());
+
+        const events = await vpn.gleanTestGetValue("interaction", "dataCollectionEnabled", "main");
+
+        assert.equal(events.length, 1);
+        var element = events[0];
+        assert.equal(element.extra.screen, "app_preferences");
+
+        //Disable data collection and check that we sent the telemetry
+        await vpn.waitForQueryAndClick(queries.screenSettings.appPreferencesView.START_AT_BOOT.visible());
+        events = await vpn.gleanTestGetValue("interaction", "dataCollectionDisabled", "main");
+
+        assert.equal(events.length, 1);
+        var element = events[0];
+        assert.equal(element.extra.screen, "app_preferences");
+    });
+
+    it("record telemetry when user clicks on Notifications in App preferences", async () => {
+        await vpn.waitForQueryAndClick(
+            queries.screenSettings.APP_PREFERENCES.visible());
+        await vpn.waitForQuery(queries.screenSettings.STACKVIEW.ready());    
+        await vpn.waitForQueryAndClick(queries.screenSettings.appPreferencesView.NOTIFICATIONS.visible());
+        await vpn.waitForQuery(queries.screenSettings.STACKVIEW.ready());
+        const events = await vpn.gleanTestGetValue("interaction", "notificationsSelected", "main");
+
+        assert.equal(events.length, 1);
+        var element = events[0];
+        assert.equal(element.extra.screen, "app_preferences");
+    });
+
+    it("record telemetry when user clicks on Language in App preferences", async () => {
+        await vpn.waitForQueryAndClick(
+            queries.screenSettings.APP_PREFERENCES.visible());
+        await vpn.waitForQuery(queries.screenSettings.STACKVIEW.ready());    
+        await vpn.waitForQueryAndClick(queries.screenSettings.appPreferencesView.LANGUAGE.visible());
+        await vpn.waitForQuery(queries.screenSettings.STACKVIEW.ready());
+
+        const events = await vpn.gleanTestGetValue("interaction", "languageSelected", "main");
+
+        assert.equal(events.length, 1);
+        var element = events[0];
+        assert.equal(element.extra.screen, "app_preferences");
+    });
+
+    it("record telemetry when user clicks on DNS settings in App preferences", async () => {
+        await vpn.waitForQueryAndClick(
+            queries.screenSettings.APP_PREFERENCES.visible());
+        await vpn.waitForQuery(queries.screenSettings.STACKVIEW.ready());    
+        await vpn.waitForQueryAndClick(queries.screenSettings.appPreferencesView.DNS_SETTINGS.visible());
+        await vpn.waitForQuery(queries.screenSettings.STACKVIEW.ready());
+
+        const events = await vpn.gleanTestGetValue("interaction", "dnsSettingsSelected", "main");
+
+        assert.equal(events.length, 1);
+        var element = events[0];
+        assert.equal(element.extra.screen, "app_preferences");
+    });
+
+    // @TODO extra is undefined here
```

Comment:
Is the whole `extra` object undefined here? Or just the `impression` extra key? That one is not set, FWIW.